Directory.Build.targets

```
@@ -24,6 +24,13 @@
     <GeneratedAssemblyInfoFile>$([System.IO.Path]::Combine('$(IntermediateOutputPath)','$(MSBuildProjectName).AssemblyInfo$(DefaultLanguageSourceExtension)'))</GeneratedAssemblyInfoFile>
   </PropertyGroup>
 
+  <!-- Language configuration -->
+  <PropertyGroup>
```

```suggestion
  <PropertyGroup Condition="'$(LangVersion)' == ''>
```

This allows overriding the `LangVersion` property in project files. That's usually the pattern we follow in target files.

---

Adding that check broke the build.  I've reverted it.  Please feel free to add it subsequently if you believe it's valuable.  Thanks.

---

Found the reason for that: https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L9.

Roslyn sets it so it's never ''. I guess providing a LangVersion default which relies on the Language property set without ignoring project overrides isn't possible.

We would need to default the LangVersion in Directory.Build.props to preview and override it in Directory.Build.targets for Language == VB to latest. I can do that in a follow-up PR if you want.